docs/UserGuide.md

```diff
@@ -42,325 +42,586 @@ Bridal Boss is a **desktop app for managing contacts, optimized for use via a  L
+- **Examples**:
+    - `deletew 2` deletes the wedding at index 2.
+    - `deletew Beach Wedding` deletes the wedding named "Beach Wedding" if there's only one match.
+- **Error Messages**:
+    - No weddings to delete:
+        - Error: "There is nothing to delete."
+    - Invalid index:
+        - Error: "The wedding index provided is invalid."
+    - Multiple matches found:
+        - System displays matching weddings and prompts for index.
 
-* Assigns the person at the specified `INDEX`. The index refers to the index number shown in the displayed person list.
-* The index **must be a positive integer** 1, 2, 3, …​
-* At least one of the optional fields must be provided.
-* When assigning roles:
-    * Each person can only have one role at a time
-    * Assigning a new role will replace the existing role
-    * Assigning blank roles e.g `r/` is not allowed
-* When assigning a person to wedding(s):
-    * The wedding(s) can be specified by `INDEX`. The index refers to the index number shown in the displayed wedding list.
-    * The index **must be a positive integer** 1, 2, 3, …​
-    * A person can be assigned to multiple weddings
+---
 
-Examples:
-* `assign 1 r/florist` assigns the 1st person to have the role of a `florist`
-* `assign 1 w/1` assigns the 1st person to the 1st wedding
-* `assign 2 r/vendor w/1 w/2` assigns the 2nd person to have the role of a `vendor` as well as to be associated with the 1st and 2nd wedding
+#### Assigning a Person: `assign`
+
+Assigns a role and/or weddings to a person.
+
+- **Formats**:
+    - **By Index**: `assign INDEX [r/ROLE] [w/WEDDING_INDEX]...`
+    - **By Name**: `assign NAME [r/ROLE] [w/WEDDING_INDEX]...`
+- **Notes**:
+    - At least one of `[r/ROLE]` or `[w/WEDDING_INDEX]` must be provided.
+    - **Role Assignment**:
+        - Each person can have only one role at a time.
+        - Assigning a new role replaces the existing role.
+        - Role cannot be blank when using `r/`.
+    - **Wedding Assignment**:
+        - Can assign a person to multiple weddings.
+        - Wedding indices must be valid and refer to existing weddings.
+        - Cannot assign the same person to the same wedding multiple times.
+        - Cannot assign a client to their own wedding as a vendor.
+- **Examples**:
+    - `assign 1 r/florist`
+        - Assigns the role "florist" to the person at index 1.
+    - `assign 1 w/1 w/2`
+        - Assigns the person at index 1 to weddings at indices 1 and 2.
+    - `assign John Doe r/photographer w/2`
+        - Assigns "John Doe" the role "photographer" and assigns to wedding at index 2.
+- **Error Messages**:
+    - Cannot assign client to own wedding:
+        - Error: "Cannot assign client to their own wedding."
+    - Person already assigned to wedding(s):
+        - Error: "Person has already been assigned to wedding(s)."
+    - Invalid wedding index:
+        - Error: "Wedding [index] is not in the list."
```

Comment:
When index is larger than size of wedding list, this error message is given "The wedding index provided is invalid, please enter an index that is between 1 and 4"

When index is 0 or negative, this is given "Index is not a non-zero unsigned integer."

So I think can change accordingly!